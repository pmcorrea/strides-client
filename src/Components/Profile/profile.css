.profile-box {
	width: 100%;
	/* border: solid black 2px; */

	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
}

.avatar {
	width: 100px;
	height: 100px;
	border-radius: 50%;
	/* background-color: gray; */
}

.stats {
	width: 90%;
	max-width: 300px;
	height: 210px;
	/* margin: .5rem auto;
	padding: .5rem; */
	margin-top: 10px;
	background-color: white;
	border-radius: .5rem;
	box-shadow: 0px 5px 10px rgb(0, 0, 0);

	display: grid;
	grid-template-columns: repeat(2, 100px);
	grid-template-rows: repeat(2, 100px);
	grid-template-areas: 
	"box-1 box-2"
	"box-3 box4";
	gap: .1rem .4rem; /* row column*/
	place-items: center start; /* column row*/
	place-content: center center; /* column row*/
}

.stats-box-header {
	color: rgb(110, 110, 110);
	font-weight: 200;
	font-size: .8rem;
}

.margin-center {
	margin: 0 auto;
}

.stats-box-num{
	text-align: center;
	color: black;
	font-size: 2rem;
}

.progress-bar-container {
	width: 90%;
	max-width: 400px;

	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
}

.progress-bar-profile {
	width: 100%;
	height: 10px;
	margin: 5px 0px;
	background-color: lightgray;
	border-radius: 5px;
}

.progress-bar-fill-profile {
	background-color: rgb(45, 209, 45);
	width: 30%;
	height: 10px;
	margin: 0px;
	border-radius: 5px;
}

.progress-bar-stats {
	display: flex;
	flex-direction: row;
	justify-content: space-between;
	margin: 0px auto;
	width: 100%;
}

.progress-bar-stats>p {
	margin: 0px;
}

.top-bar {
	width: 100%;
	color: white;
	/* background-color: gray; */
	/* padding: 10px; */
	margin-bottom: 10px;

	display: flex;
	flex-direction: row;
	justify-content: space-between;
	align-items: center;
}

.top-bar>div {
	margin: 0px;
}

.top-bar>button {
	margin: 0px;
}