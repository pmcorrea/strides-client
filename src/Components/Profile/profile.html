<!DOCTYPE html>
<html lang="en">

	<head>
		<meta charset="UTF-8">
		<meta name="viewport" content="width=device-width, initial-scale=1.0">
		<link rel="stylesheet" href="../Login/index.css" type="text/css">
		<link rel="stylesheet" href="../Home/home.css" type="text/css">
		<link rel="stylesheet" href="../Habit/habit.css" type="text/css">
		<link rel="stylesheet" href="profile.css" type="text/css">
		<title>Strides</title>
	</head>

	<body>

		<div class="profile-box">
			<div class="top-bar">
				<button class="circle-button">Logo</button>
				<button><img src="../Assets/gear.svg" alt="" class="icon-bar-icon"></button>
			</div>


			<img src="../Assets/avatar.jpg" alt="" class="avatar">


			<h2>Penelope Arias</h2>

			<div class="progress-bar-container">
				<div class="progress-bar-profile">
					<div class="progress-bar-fill-profile"></div>
				</div>

				<div class="progress-bar-stats">
					<p>lvl 4</p>
					<p>26%</p>
				</div>
			</div>

			<div class="stats">
				<div class="box-1">
					<p class="stats-box-header">habits done</p>
					<p class="stats-box-num">9</p>
				</div>

				<div class="box-2">
					<p class="stats-box-header">logged days</p>
					<p class="stats-box-num">9</p>
				</div>

				<div class="box-3">
					<p class="stats-box-header">perfect habits</p>
					<p class="stats-box-num">9</p>
				</div>
				
				<div class="box-4">
					<p class="stats-box-header">biggest streak</p>
					<p class="stats-box-num">9</p>
				</div>
			</div>
		</div>

		<div class="bottom-toolbar">
			<button><img src="../Assets/home-icon.svg" alt="" class="icon-bar-icon"></button>
			<button><img src="../Assets/plus-icon.svg" alt="" class="icon-bar-icon"></button>
			<button><img src="../Assets/profile-icon.svg" alt="" class="icon-bar-icon"></button>
		</div>
	</body>

</html>