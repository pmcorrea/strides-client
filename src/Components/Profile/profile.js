import React, { useEffect } from 'react';
import { Link } from "react-router-dom";
import ErrBoundary from "../ErrBoundary/err-boundary"
import "./profile.css"
import Gear from "../../Assets/settings.svg"
import Avatar from 	"../../Assets/avatar.svg"
import Logout from "../../Assets/logout.svg"

import HomeIcon from "../../Assets/home-icon.svg"
import PlusIcon from "../../Assets/plus-icon.svg"
import ProfileIcon from "../../Assets/profile-icon.svg"

import { useQuery } from '@apollo/client'
import { userById } from "../../Queries/queries"
import TokenHelpers from "../../Services/token-helpers"
import { useApolloClient } from '@apollo/client'

export default function Profile() {
	const { loading, data, refetch } = useQuery(userById)
	const client = useApolloClient()

	useEffect(() => {
		refetch()
	})
	
	if (loading) {
		return (" ")
	}

	function logOut() {
		TokenHelpers.clearAuthToken()
		client.resetStore()
	}

	return (
	<div>
			<ErrBoundary>
		<div className="profile-box">
			<div className="top-bar">
				<button>
					<Link to="/settings">
						<img src={Gear} alt="" className="icon-bar-icon" />
					</Link>
				</button>

				<Link to="/">
					<button className="logout-button" onClick={() => logOut()}>
						<img src={Logout} alt="" className="icon-bar-icon logout-button" />
					</button>
				</Link>	
			</div>


			<img src={Avatar} alt="" className="avatar" />
			<h2>{data["userById"][0]["user_name"]}</h2>

			<div className="stats">
				<div className="box-1 margin-center">
					<p className="stats-box-header">habits done</p>
					<p className="stats-box-num">{data["userById"][0]["habits_done"]}</p>
				</div>

				<div className="box-2 margin-center">
					<p className="stats-box-header">logged days</p>
					<p className="stats-box-num">{data["userById"][0]["logged_total"]}</p>
				</div>

				<div className="box-3 margin-center">
					<p className="stats-box-header">perfect habits</p>
					<p className="stats-box-num">{data["userById"][0]["perfect_habits"]}</p>
				</div>

				<div className="box-4 margin-center">
					<p className="stats-box-header">biggest streak</p>
					<p className="stats-box-num">{data["userById"][0]["biggest_streak"]}</p>
				</div>
			</div>
		</div>

		<div className="bottom-toolbar">
			<button>
				<Link to="/home">
					<img src={HomeIcon} alt="" className="icon-bar-icon" />
				</Link>
			</button>

			<button>
				<Link to="/add-habit">
					<img src={PlusIcon} alt="" className="icon-bar-icon" />
				</Link>
			</button>

			<button>
				<Link to="/profile">
					<img src={ProfileIcon} alt="" className="icon-bar-icon" />
				</Link>
			</button>
		</div>	
			</ErrBoundary>
	</div>
	);
}