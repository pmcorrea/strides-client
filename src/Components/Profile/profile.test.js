import React from 'react';
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect'
import Profile from './profile';

afterEach(cleanup)

// Snapshot Test
it ('should take a snapshot', () => {
	const { asFragment } = render(
	<BrowserRouter>
			<Profile />
	</BrowserRouter>
	)

	expect(asFragment(
		<BrowserRouter>
			<Profile />
		</BrowserRouter>)
		).toMatchSnapshot()
})

// Smoke Test
it('renders without crashing', () => {
	const div = document.createElement('div')

	ReactDOM.render(
		<BrowserRouter>
			<Profile />
		</BrowserRouter>,
		div
	)
	
	ReactDOM.unmountComponentAtNode(div)
})