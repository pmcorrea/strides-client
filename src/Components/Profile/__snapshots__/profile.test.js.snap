// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="profile-box"
    >
      <div
        class="top-bar"
      >
        <button
          class="circle-button"
        >
          Logo
        </button>
        <button>
          <a
            href="/settings"
          >
            <img
              alt=""
              class="icon-bar-icon"
              src="gear.svg"
            />
          </a>
        </button>
      </div>
      <img
        alt=""
        class="avatar"
        src="avatar.jpg"
      />
      <h2>
        Penelope Arias
      </h2>
      <div
        class="progress-bar-container"
      >
        <div
          class="progress-bar-profile"
        >
          <div
            class="progress-bar-fill-profile"
          />
        </div>
        <div
          class="progress-bar-stats"
        >
          <p>
            lvl 4
          </p>
          <p>
            26%
          </p>
        </div>
      </div>
      <div
        class="stats"
      >
        <div
          class="box-1 margin-center"
        >
          <p
            class="stats-box-header"
          >
            habits done
          </p>
          <p
            class="stats-box-num"
          >
            9
          </p>
        </div>
        <div
          class="box-2 margin-center"
        >
          <p
            class="stats-box-header"
          >
            logged days
          </p>
          <p
            class="stats-box-num"
          >
            9
          </p>
        </div>
        <div
          class="box-3 margin-center"
        >
          <p
            class="stats-box-header"
          >
            perfect habits
          </p>
          <p
            class="stats-box-num"
          >
            9
          </p>
        </div>
        <div
          class="box-4 margin-center"
        >
          <p
            class="stats-box-header"
          >
            biggest streak
          </p>
          <p
            class="stats-box-num"
          >
            9
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
